pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ObjectFunctions.java

```diff
@@ -94,33 +94,41 @@ private static Object coalesceVar(Object... objects) {
     return null;
   }
 
-  @ScalarFunction
-  public static Object caseWhen(boolean c1, Object o1, Object oe) {
+  @Nullable
+  @ScalarFunction(nullableParameters = true, names = {"case", "caseWhen", "case_when"})
```

Comment:
technically `caseWhen` is not supposed to be a valid query function (e.g. should not be user-facing). It is only done this way b/c in V2 engine we manually encode the `case when ... then .. else ... end` as `caseWhen` to distinguish from the function name `case`, which is matched to the transform function instead of the scalar function

Reply:
Our issue was `case when...then..else..end` with only literals was sent to the transform function resulting in multiple rows of same literal result. When we added name 'case', it was correctly handled only in broker and returned 1 result. This was in v1 engine. Would it be treated as case_when if we used v2?

Reply:
I see. so basically IIUC --> 
- your expectation of `SELECT CASE WHEN 2 > 0 THEN 'YES' ELSE 'NO' END FROM tbl` is to return a single row with `"YES"`, but now it returns multiple rows of `"YES"` (in fact no. of row in tbl)
- you don't care about whether `CASEWHEN` syntax. 

is that correct?

Reply:
yes. we would like CASE WHEN to go to compile time invoker when theres only literals and just go to transform function as usual otherwise. 
We are able to get this with only adding the name case. Should we remove the names caseWhen and case_when?

Reply:
ok got it. this PR should be good for achieving this goal and it doesn't have any effect on caseWhen as it is not suppose to be directly used anyway